mmrazor/models/pruners/ratio_pruning.py

```diff
@@ -26,52 +28,183 @@ class RatioPruner(StructurePruner):
+
+        for module_name in self.modules_have_ancest:
+            module = self.name2module[module_name]
+            parents = self.node2parents[module_name]
+            # To avoid ambiguity, we only allow the following two cases:
+            # 1. all elements in parents are ``Conv2d``,
+            # 2. there is only one element in parents, ``concat`` or ``chunk``
+            # In case 1, all the ``Conv2d`` share the same space_id and
+            # out_mask.
+            # So in all cases, we only need the very first element in parents
+            parent = parents[0]
+            space_id = self.get_space_id(parent)
+
+            if isinstance(space_id, dict):
+                if 'concat' in space_id:
+                    in_mask = []
+                    for parent_space_id in space_id['concat']:
+                        parent_out_mask = self.channel_bin2channel(
+                            subnet_dict[parent_space_id],
+                            self.space_id2out_mask[parent_space_id])
+                        in_mask.append(parent_out_mask)
+                    module.in_mask = torch.cat(
+                        in_mask, dim=1).to(module.in_mask.device)
+            else:
+                parent_out_mask = self.channel_bin2channel(
+                    subnet_dict[space_id], self.space_id2out_mask[space_id])
+                module.in_mask = parent_out_mask.to(module.in_mask.device)
+
     def set_min_channel(self):
         """Set the number of channels each layer to minimum."""
         subnet_dict = dict()
-        for space_id, out_mask in self.channel_spaces.items():
-            out_channels = out_mask.size(1)
-            random_ratio = self.min_ratio
-            new_channels = int(round(out_channels * random_ratio))
-            assert new_channels > 0, \
-                'Output channels should be a positive integer.'
-            new_out_mask = torch.zeros_like(out_mask)
-            new_out_mask[:, :new_channels] = 1
-
-            subnet_dict[space_id] = new_out_mask
+        for space_id in self.search_spaces.keys():
+            subnet_dict[space_id] = 1
+        self.set_subnet(subnet_dict)
 
+    def set_max_channel(self):
+        """Set the number of channels each layer to maximum."""
+        subnet_dict = dict()
+        for space_id in self.search_spaces.keys():
+            subnet_dict[space_id] = self.channel_bins
         self.set_subnet(subnet_dict)
 
+    def get_max_channel_bins(self, max_channel_bins=None):
+        """Get the max number of channel bins of all the groups which can be
+        pruned during searching."""
+        # deprecate arguments warning
+        warnings.warn(
+            'The useless argument ``max_channel_bins`` will be deprecated.')
+        subnet_dict = dict()
+        for space_id in self.search_spaces.keys():
+            subnet_dict[space_id] = self.channel_bins
+        return subnet_dict
+
+    def build_search_spaces(self):
```

Comment:
The function overrides method of parent class but only add a new attribute `space_id2out_mask`, is it possible to simplify with the overrided method?